ext/opentelemetry-ext-grpc/src/opentelemetry/ext/grpc/__init__.py

```diff
@@ -16,9 +16,62 @@
 # pylint:disable=import-self
 # pylint:disable=no-name-in-module
 # pylint:disable=relative-beyond-top-level
+from contextlib import contextmanager
 
+import grpc
+from concurrent import futures
 from opentelemetry import trace
 from opentelemetry.ext.grpc.version import __version__
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.auto_instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.ext.grpc.grpcext import intercept_channel, intercept_server
+
+
+class GrpcInstrumentorServer(BaseInstrumentor):
+    trace.set_tracer_provider(TracerProvider())
+    tracer = trace.get_tracer_provider()
+    server = None
+
+    def _instrument(self, **kwargs):
+        self.server = grpc.server(futures.ThreadPoolExecutor())
+        self.server = intercept_server(self.server, server_interceptor())
```

Comment:
We should be wrapping `grpc.server` here instead of calling it to allow for auto instrumentation to work properly. Use `wrapt`, ex: [dbapi wrapper function](https://github.com/open-telemetry/opentelemetry-python/blob/master/ext/opentelemetry-ext-dbapi/src/opentelemetry/ext/dbapi/__init__.py#L128)